plugins/chat/app/services/chat/list_channel_messages.rb

```diff
@@ -170,5 +171,12 @@ def update_membership_last_viewed_at(guardian:)
         context.membership&.update!(last_viewed_at: Time.zone.now)
       end
     end
+
+    def update_user_last_channel(guardian:, channel:)
```

Comment:
We store the last channel id each time you load messages for this channel

Reply:
This is nice, should be more efficient than loading every time 👍